Aws/S3/Commands/Multipart.hs

```diff
@@ -131,12 +131,13 @@ data UploadPart = UploadPart {
   , upContentMD5 :: Maybe (Digest MD5)
   , upServerSideEncryption :: Maybe ServerSideEncryption
   , upRequestBody  :: HTTP.RequestBody
+  , upExpect100Continue :: Bool -- ^ Note: Requires http-client >= 0.4.10
```

Comment:
I'm not sure. I don't think it is always desired for uploads but rather an additional safeguard that can be enabled. It is especially relevant for streaming cases from non-repeatable sources, as I outlined in the `http-client` PR. Delaying the body comes at a certain cost in form of an obvious additional delay for the whole upload as well as two distinct response timeouts in the case of `http-client`, so if you have a repeatable source (and maybe even small payloads), the extra overhead might not be worth it, since the request will be successfully retried for the occasional encounter of a closed connection or a redirect, with the only disadvantage of sinking parts of the body into the wrong request / connection before detecting the error.

It might be worth noting that enabling this feature with an http-client version < 0.4.10 won't cause any unexpected issues, it will just not have the desired effect of delaying the body until the `100 Continue`, hence I thought a note is good enough and there is no need for conditional compilation or even an `http-client >= 0.4.10` cabal constraint.